### PR TITLE
Unit tests store tenant

### LINKED ISSUE #147: Add / update unit test for tenant store
**Why**
Increase our test coverage and give us increased confidence in the tenant store

**What**
* [ ] Test for state
* [ ] Test getters
* [ ] Test mutations
* [ ]  Test actions

**Notes**
Sub issue of #101 
Some work done in cloud-web-ui but will need updates

### PR DESCRIPTION
**Pull request by zhen0**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [x] add/update tests (or don't, for reasons explained below)

## Describe this PR
- Adds unit tests for the tenant Vuex store
- Closes #147

### CODE REVIEW

File: tests/unit/store/tenant.spec.js [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+        mockerror = false
+        store = new Vuex.Store({
+          state: initialTenantState(),
+          getters: tenant.getters,
+          mutations: tenant.mutations,
+          actions: tenant.actions
+        })
+      })
+
+      it('should set tenants', async () => {
```
Comment (znicholasbrown) [inline tests/unit/store/tenant.spec.js:240]:
```suggestion
      it('should call the prefectTenants method', async () => {
```

File: tests/unit/store/tenant.spec.js [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+      it('should return tenants', async () => {
+        const tenantsArray = [{ name: 'boo', id: '12345' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        const returnedTenants = await store.dispatch('getTenants')
+        expect(returnedTenants).toEqual([{ name: 'boo', id: '12345' }])
+      })
+    })
+
+    describe('setCurrentTenant -server', () => {
```
Comment (znicholasbrown) [inline tests/unit/store/tenant.spec.js:260]:
```suggestion
    describe('setCurrentTenant - Server', () => {
```

File: tests/unit/store/tenant.spec.js [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        expect(await store.dispatch('setCurrentTenant', 'team2')).toEqual({
+          id: '12345',
+          name: 'boo',
+          role: 'TENANT_ADMIN',
+          slug: 'team2'
+        })
+      })
+    })
+
+    describe('setCurrentTenant - Cloud specific', () => {
```
Comment (znicholasbrown) [inline tests/unit/store/tenant.spec.js:315]:
```suggestion
    describe('setCurrentTenant - Cloud', () => {
```

File: tests/unit/store/tenant.spec.js [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+        })
+      })
+
+      it('should throw an error if no slug is provided', () => {
+        expect(store.dispatch('setCurrentTenant', false)).rejects.toThrow(
+          'No slug was provided when trying to set the current tenant'
+        )
+      })
+
+      it('should check if tenant is set and getTenant if not set', async () => {
```
Comment (znicholasbrown) [inline tests/unit/store/tenant.spec.js:278]:
```suggestion
      it('should call getTenants if the passed tenant is not present in the store', async () => {
```

File: tests/unit/store/tenant.spec.js [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+        expect(store.getters.tenant.slug).toBe(null)
+        const tenantsArray = [{ name: 'boo', id: '12345', slug: 'team2' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        await store.dispatch('setCurrentTenant', 'team2')
+        expect(store.getters.tenant.slug).toEqual('team2')
```
Comment (znicholasbrown) [inline tests/unit/store/tenant.spec.js:283]: Let's test this against the entire tenant object described in `tenantsArray`

File: tests/unit/store/tenant.spec.js [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+      it('should throw an error if the requested tenant does not exist', async () => {
+        expect(store.getters.tenant.slug).toBe(null)
+        const tenantsArray = [{ name: 'boo', id: '12345', slug: 'team2' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        expect(store.dispatch('setCurrentTenant', 'team3')).rejects.toThrow(
+          "Error: Unable to set current tenant: tenant doesn't exist"
+        )
+      })
+
+      it('should set the tenant role to Tenant Admin', async () => {
```
Comment (znicholasbrown) [inline tests/unit/store/tenant.spec.js:295]:
```suggestion
      it('should set the tenant role to TENANT_ADMIN', async () => {
```

Comment (znicholasbrown) [reply tests/unit/store/tenant.spec.js:295]: Can you add a comment describing why this is the case? It'll be useful when debugging this for someone unfamiliar with the differences between Server and Cloud.

File: tests/unit/store/tenant.spec.js [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+            'user/memberships': () => [
+              { tenant: { id: '12345' }, role: 'USER' }
+            ]
+          },
+          mutations: tenant.mutations,
+          actions: tenant.actions
+        })
+      })
+
+      it('should set the tenant role according to user memberships', async () => {
```
Comment (znicholasbrown) [inline tests/unit/store/tenant.spec.js:335]: Maybe this should be a sub-describe block that tests each of the 3 roles?

File: tests/unit/store/tenant.spec.js [hunk 8]
```diff
@@ -0,0 +1,369 @@
...
+          actions: tenant.actions
+        })
+      })
+
+      it('should throw an error if no settings are given', async () => {
+        expect(store.dispatch('updateTenantSettings')).rejects.toThrow(
+          'Error: passed invalid or empty settings object'
+        )
+      })
+
```
Comment (znicholasbrown) [inline tests/unit/store/tenant.spec.js:362]: Seems like we could add another test here for the update tenant settings mutation:
```
it ('should call the UpdateTenantSettings mutation with the passed settings', async () => {
  expect(mockedUpdateTenantSettings).toHaveBeenCalledWith({ variables: { name: 'Jerry' } })
})
```

**Review by znicholasbrown — CHANGES_REQUESTED** [review]
Overall looks good @zhen0 - I've added some comments and suggestions.

**Comment by zhen0** [comment]
Thanks for the comments @znicholasbrown - ready for another review

**Review by znicholasbrown — APPROVED** [review]
Thanks for the changes @zhen0 !

**Pull request merged**